### PR TITLE
[32] [34] [35] [38] [41] [Integrate] Receive notifications when mentioned, review requested, reviewed, assigned

### PR DESCRIPTION
**Pull request by llleyelll**
#32 
#34 
#35 
#38 
#41 

## What happened 👀

Schedule notification when:
- Got mentioned
- Review requested
- Someone reviewed your PR
- You were assigned
 
## Insight 📝

- Currently, CI updated only support for email and website.
- This PR included all tickets for receive notifications as each notification type only has a small implementation.
- Fix recent notification repeatly showing 
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/45258998/134453395-2bb77390-9451-4cb8-813e-1e08648f9b3d.mov

Screen Shot 2021-09-23 at 10 26 09 AM

### CODE REVIEW

File: Source/Application/AppDelegate/AppDelegate.swift [hunk 1]
```diff
@@ -49,8 +49,26 @@ extension AppDelegate {
...
-            .sink(receiveValue: { models in
-                #warning("Show notification for received models")
+            .sink(receiveValue: { notifications in
+                notifications.forEach {
+                    var isShowNotiBanner: Bool = false
+                    switch $0.reason {
+                    case .assign, .mention, .review_requested:
+                        isShowNotiBanner = true
+                    case .author:
+                        isShowNotiBanner = $0.subject.type == "PullRequest" ? true : false
```
Comment (blyscuit) [inline Source/Application/AppDelegate/AppDelegate.swift:59]: Please add this property to APINotification.Subject

**Review by blyscuit — APPROVED** [review]

**Review by markgravity — APPROVED** [review]
LGTM

**Comment by blyscuit** [comment]
@llleyelll please rebase to with `scheduleNotification` argument.

**Comment by llleyelll** [comment]
> @llleyelll please rebase to with `scheduleNotification` argument.

@blyscuit I updated the code, please help check 🙇‍♀️

**Review by blyscuit — APPROVED** [review]

**Pull request merged**